src/docs/arch-row.dox

```diff
@@ -1,12 +1,73 @@
-/*! @arch_page arch-row Row Store
+/*! @arch_page arch-row-column Row Store & Column Store
 
-Row Stores are Btrees stored in WiredTiger that do not have a record id
-as key.  Thus, they implement a generalized version of a Btree, where
-the key and data can be arbitrary length.
+@section Definition
+
+A row oriented storage is a database management system where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column. WiredTiger supports both and each management system is
+represented by a certain type of B-Tree data structure.
+
+Keys and values in row-stores can be anything of arbitrary length, thus, row-stores implement a
+generalized version of a B-Tree. Furthermore, keys are explicitly stored. Column-stores use a 64-bit
+unsigned integer record id as keys, thus, column-stores implement a specialized version of a B-Tree
+where the key has a predictable length. In the case of column-stores, keys are implied given the
+B-Tree position.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to 1 and 8 bits in length. The bit length is specified when the column store
+is created. The fixed length column store has specialized use cases like bitmaps.
+- The variable length is a more general case which allows for values to have any length similarly to
+row-stores.
+
+@section Main differences
```

Comment:
@keithbostic, let me know if the two following paragraphs make more sense. I have tried to reword both of them addressing your comments.